src/Promotions/InPluginUpsells/ProductRecommendationsRoute.php

```diff
@@ -0,0 +1,112 @@
+    {
+        register_rest_route(
+            'give-api/v2',
+            $this->endpoint,
+            [
+                [
+                    'methods' => ['POST'],
+                    'callback' => [$this, 'handleRequest'],
+                    'permission_callback' => [$this, 'permissionsCheck'],
+                    'args' => [
+                        'option' => [
+                            'type' => 'string',
+                            'required' => true,
+                            'enum' => [
+                                'givewp_donations_recurring_recommendation_dismissed',
+                                'givewp_donations_fee_recovery_recommendation_dismissed',
+                                'givewp_donations_designated_funds_recommendation_dismissed',
+                            ],
+                        ],
+                    ],
+                ],
+            ]
+        );
+    }
+
+    /**
+     * @unreleased
+     */
+    public function permissionsCheck()
+    {
+        if ( ! current_user_can('manage_options')) {
+            return new WP_Error(
+                'rest_forbidden',
+                esc_html__('You don\'t have permission to dismiss options. Only users with the "manage_options" capability can perform this action.',
+                    'give'),
+                ['status' => $this->authorizationStatusCode()]
+            );
+        }
+
+        return true;
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param WP_REST_Request $request
+     *
+     * @return WP_REST_Response
+     */
+    public function handleRequest(WP_REST_Request $request)
+    {
+        $errors = [];
+        $successes = [];
+
+        try {
+            switch ($request->get_param('option')) {
+                case 'givewp_donations_recurring_recommendation_dismissed':
+                    update_option('givewp_donations_recurring_recommendation_dismissed', true);
+                    $successes[] = 'givewp_donations_recurring_recommendation_dismissed';
+                    break;
+
+                case 'givewp_donations_fee_recovery_recommendation_dismissed' :
+                    update_option('givewp_donations_fee_recovery_recommendation_dismissed', true);
+                    $successes[] = 'givewp_donations_fee_recovery_recommendation_dismissed';
+                    break;
+
+                case 'givewp_donations_designated_funds_recommendation_dismissed':
+                    update_option('givewp_donations_designated_funds_recommendation_dismissed', true);
+                    $successes[] = 'givewp_donations_designated_funds_recommendation_dismissed';
+                    break;
+
+                default:
+                    $errors[] = 'Invalid option';
+            }
+        } catch (\Exception $e) {
+            $errors[] = $e->getMessage();
+        }
+
+        if (count($errors) > 0) {
+            $response = new WP_Error('invalid_option', __('Invalid option'), ['status' => 400]);
```

Comment:
We're storing up the errors but then never actually using them except to check if there are any. Also, we should be using `Log::error()` to track this, which should include the errors.

Reply:
I sent a log with the response and errors array.  Resolved: https://github.com/impress-org/givewp/pull/6797/commits/fe128881ab302c6f729eca5b330ba98fd45d8aec If you open the commit you'll see it was failing. I have since addressed the failed checks in the commits 3 commits after.